static/src/javascripts/projects/commercial/modules/dfp/dfp-api.spec.js

```diff
@@ -0,0 +1,422 @@
+    `;
+
+    beforeEach(done => {
+        config.switches = {
+            commercial: true,
+            sonobiHeaderBidding: false,
+        };
+        config.page = {
+            adUnit: '/123456/theguardian.com/front',
+            contentType: 'Article',
+            edition: 'us',
+            isFront: true,
+            keywordIds: 'world/korea,world/ukraine',
+            pageId: 'world/uk',
+            section: 'news',
+            seriesId: 'learning/series/happy-times',
+            sharedAdTargeting: {
+                ct: 'Article',
+                edition: 'us',
+                k: ['korea', 'ukraine'],
+                se: ['happy-times'],
+            },
+        };
+        config.images = {
+            commercial: {},
+        };
+        config.ophan = {
+            pageViewId: 'dummyOphanPageViewId',
+        };
+
+        if (document.body) {
+            document.body.innerHTML = domSnippet;
+        }
+
+        $style = $.create('<style type="text/css"></style>')
+            .html(`body:after{ content: "wide"}`)
+            .appendTo('head');
+        const pubAds = {
+            listeners: [],
+            addEventListener: jest.fn(function(eventName, callback) {
+                this.listeners[eventName] = callback;
+            }),
+            setTargeting: jest.fn(),
+            enableSingleRequest: jest.fn(),
+            collapseEmptyDivs: jest.fn(),
+            refresh: jest.fn(),
+        };
+        const sizeMapping = {
+            sizes: [],
+            addSize: jest.fn(function(width, sizes) {
+                this.sizes.unshift([width, sizes]);
+            }),
+            build: jest.fn(function() {
+                const tmp = this.sizes;
+                this.sizes = [];
+                return tmp;
+            }),
+        };
+        window.googletag = {
+            cmd: {
+                push(...argz) {
+                    const args = Array.prototype.slice.call(argz);
+                    args.forEach(command => {
+                        command();
+                    });
+                },
+            },
+            pubads() {
+                return pubAds;
+            },
+            sizeMapping() {
+                return sizeMapping;
+            },
+            defineSlot: jest.fn(() => window.googletag),
+            defineOutOfPageSlot: jest.fn(() => window.googletag),
+            addService: jest.fn(() => window.googletag),
+            defineSizeMapping: jest.fn(() => window.googletag),
+            setTargeting: jest.fn(() => window.googletag),
+            enableServices: jest.fn(),
+            display: jest.fn(),
+        };
+        // eslint-disable-next-line no-underscore-dangle
+        window.__switch_zero = false;
+
+        commercialFeatures.dfpAdvertising = true;
+
+        done();
```

Comment:
Why does this require a `done()` function? All of the code looks very sync to me :)